analyzere/requestor.py

```diff
@@ -33,21 +33,28 @@ def handle_api_error(resp, code):
         raise errors.ServerError(message, body, code, json_body)
 
 
-def request(method, path, params=None, data=None, auto_retry=True):
+def request(method, path, params=None, headers=None, data=None,
+            auto_retry=True):
     """
     method - HTTP method. e.g. get, put, post, etc.
     path - Path to resource. e.g. /loss_sets/1234
     params - Parameter to pass in the query string
+    headers - Dictionary of additional HTTP headers
     data - Dictionary of parameters to pass in the request body
     """
     body = None
     if data is not None:
         body = json.dumps(data, cls=utils.DateTimeEncoder)
 
-    headers = {
-        'accept': 'application/json',
-        'content-type': 'application/json',
-    }
+    # append the default set of headers to any user provided headers
```

Comment:
sorry to pick on this, but it's a common pattern in python to handle optional dictionaries as follows:

```
def whatever_method(default_dict=None):
    default_dict = default_dict or {}
    default_dict.update({
        'accept': 'application/json',
        'content-type': 'application/json',
    })
```